### PR TITLE
fix(globals): Resolve exported globals dynamically.

### PR DESCRIPTION
**Pull request by mgiambalvo**
Because globals are set in the runner, they may be undefined at import time. Also, the `browser` global object will change if `browser.reset()` is called. By making these globals exported as properties, they will be resolved dynamically. This also fixes issues importing protractor in config files written in typescript, since the globals are resolved at call time instead of at import time.

Note that until the Runner calls `setupGlobals_`, these exports will be undefined. That's just the nature of these objects, though, and the global objects have the same behavior.

### CODE REVIEW

File: spec/basic/restart_spec.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+fdescribe('browser.restart', function() {
+  fit('doesn\'t break ignoreSynchronization', function() {
+    browser.get('index.html#/polling');
+    browser.restart();
+
+    browser.ignoreSynchronization = true;
+    // Get a non-angular page. It shouldn't fail if ignoreSynchronization is on.
+    browser.get("https://google.com/");
```
Comment (cnishina) [inline spec/basic/restart_spec.js]: Nit: should use single quotes.

File: spec/basic/restart_spec.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+fdescribe('browser.restart', function() {
```
Comment (cnishina) [inline spec/basic/restart_spec.js]: Probably do not need fdescribe and fit

**Comment by mgiambalvo** [comment]
Also fixes #3605

**Comment by KrauseStefan** [comment]
Great work, happy that this made it in. :thumbsup:
This solved my issue #3627 thank you

**Pull request merged**